Task 2

MariaNovozhilova opened the pull request:
Иван, привет, из правок что сделала:

доступы вынесены в jsons
вернула снова return connection в функции create_connection, так как потом нужно сделать commit и close в соединении и с курсором это не получится

и что не сделала:

-не чистила stdout 
-не логировала количество строк для проверки
так как до 12 я не успела бы тогда, я оч медленно пишу

mgorshkov committed:
Python-script to copy data from source layer into ODS layer 'as is'.
This version is prepared to copy just one table.

MariaNovozhilova committed:
Script for copying data from source tables into ods layer.
Iterates the list of tables.

Postnikovia commented on script_1.18.py:
```diff
@@ -0,0 +1,51 @@
+    except psycopg.OperationalError as e:
+        print(f"The error '{e}' occurred")
+    return connection
+
```
Внести все в одну функцию

Postnikovia replied:
решено

MariaNovozhilova committed:
Update script_1.18.py

correct two tables names
updated for-cycle - added re-connection if connection has closed after previous step

MariaNovozhilova committed:
Update script_1.18.py

the whole script wrapped into func

MariaNovozhilova committed:
Update script_1.18.py

indentation fixed

MariaNovozhilova committed:
Update script_1.18.py

add truncate target table before copy

MariaNovozhilova committed:
SQL script for lgs layer views

Postnikovia committed:
Update script_1.18.py

причесал немного функиции так должно быть более читаемо
но я не проверял работаспособность кода(нужно проверить и заново выслать МР)

Функция в функции -  не самое оптимальное решение, исключение если используется замыкание(можно отдельно почитать)
Основная функция -  функция которая последовательно запускает последовательность ранее созданных функций

MariaNovozhilova committed:
python with sql script for cleaning data and loading
it into dds layer

MariaNovozhilova committed:
Актуальные скрипты

MariaNovozhilova committed:
Merge branch 'Task-2' of https://github.com/MariaNovozhilova/DAR_team7 into Task-2

MariaNovozhilova committed:
Rename Script_ODS to Script_ODS_draft

this script is not ready and is not supposed to be here

MariaNovozhilova committed:
SQL script for DM_tables
Python&SQL to fill DM tables with data
Python&SQL for collecting bad data
Python&SQL for data cleaning and filing dds layer

MariaNovozhilova committed:
Merge branch 'Task-2' of https://github.com/MariaNovozhilova/DAR_team7 into Task-2

MariaNovozhilova committed:
some column names updated
table total_certificates deleted as long as the data included into
personal data

Postnikovia commented on CREATE view lgs.txt:
Все хорошо

Postnikovia commented on DM_tables.py:
Цикл бы конечно в отдельную функцию внести чтобы потом удобнее было запускать из  airflow

Postnikovia commented on DM_views.py:
Скорее всего неверное расширение файла (на sql похож)

MariaNovozhilova replied:
да, это уже не актуальный файл, я переделала на DM_tables, и там уже полный скрипт

Postnikovia commented on Script_ODS_draft:
```diff
@@ -0,0 +1,48 @@
+
+# цикл для перебора таблиц
+#  
+for table in tables:
```
тут задачи для airflow создаются в цикле а в 46 и 48 строках  последовательность шагов задается для одного шага( по сути добавит связь только с последними шагами)
Нужно либо создавать зависимости внутри цикла
Либо добавить шаги в список и создать зависимость к списку: start_step >> [extract_step] >> [insert_step]
 >> end_step

Postnikovia commented on requirements.txt:
файл пустой

Postnikovia commented on from lgc to dds with cleaning.py:
```diff
@@ -0,0 +1,375 @@
+, CAST(regexp_replace(ldael.dbms, '[^0-9]', '', 'g') AS INTEGER)
+FROM lgc.dbms_and_employee_grade AS ldael;
+UPDATE dds.dbms_and_employee_grade
+SET sort = CAST(CASE WHEN grade = '115637' THEN '100'
```
Есть ощущение что это дело можно через джойны написать,  потому что если данные в БД поменяются придется скрипт переписывать,  а это не очень

Postnikovia commented on from lgc to dds with cleaning.py:
```diff
@@ -0,0 +1,375 @@
+, user_id
+, CAST(CASE WHEN active = 'Да' THEN 'True'
+			WHEN active = 'Нет' THEN 'False' END AS BOOL)
+, NULLIF(dbms, '')
```
Не очень понял зачем эта конструкция

MariaNovozhilova committed:
Merge branch 'Maria_tasks' into Task-2
